xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-authentication/xwiki-platform-security-authentication-default/src/test/java/org/xwiki/security/authentication/internal/DefaultResetPasswordManagerTest.java

```diff
@@ -464,150 +466,199 @@ void resetPasswordNotDocumentReferenceUser() throws Exception
-        assertFalse(this.resetPasswordManager.isPasswordCompliantWithRegistrationRules("_/3434"));
-
-        when(this.registrationConfiguration.getPasswordMinimumLength()).thenReturn(13);
-        assertFalse(this.resetPasswordManager.isPasswordCompliantWithRegistrationRules("foo"));
-        assertFalse(this.resetPasswordManager.isPasswordCompliantWithRegistrationRules("fooooo"));
-        assertFalse(this.resetPasswordManager.isPasswordCompliantWithRegistrationRules("foooOOOOOoo"));
-        assertFalse(this.resetPasswordManager.isPasswordCompliantWithRegistrationRules("foo000"));
-        assertFalse(this.resetPasswordManager.isPasswordCompliantWithRegistrationRules("fooOOOOo000"));
-        assertFalse(this.resetPasswordManager.isPasswordCompliantWithRegistrationRules("foo_/ 000"));
-        assertFalse(this.resetPasswordManager.isPasswordCompliantWithRegistrationRules("foo_/"));
-        assertFalse(this.resetPasswordManager.isPasswordCompliantWithRegistrationRules("fOOOOoo_/"));
-        assertFalse(this.resetPasswordManager.isPasswordCompliantWithRegistrationRules("foOOOo_/ 000"));
-        assertTrue(this.resetPasswordManager.isPasswordCompliantWithRegistrationRules("foOOOo_/ éé000"));
-        assertFalse(this.resetPasswordManager.isPasswordCompliantWithRegistrationRules(" _/"));
-        assertFalse(this.resetPasswordManager.isPasswordCompliantWithRegistrationRules("_/3434"));
+        when(this.registrationConfiguration.getPasswordMinimumLength()).thenReturn(minimumLength);
+        when(this.registrationConfiguration.getPasswordRules()).thenReturn(rules);
+        assertEquals(expectation, this.resetPasswordManager.isPasswordCompliantWithRegistrationRules(password));
+    }
+
+    private static Stream<Arguments> provideArgumentsForIsPasswordCompliantWithRegistrationRules()
+    {
+        Set<RegistrationConfiguration.PasswordRules> noRules = Set.of();
+        Set<RegistrationConfiguration.PasswordRules> lowerMandatory =
+            Set.of(RegistrationConfiguration.PasswordRules.ONE_LOWER_CASE_CHARACTER);
+        Set<RegistrationConfiguration.PasswordRules> upperMandatory =
+            Set.of(RegistrationConfiguration.PasswordRules.ONE_UPPER_CASE_CHARACTER);
+        Set<RegistrationConfiguration.PasswordRules> symbolMandatory =
+            Set.of(RegistrationConfiguration.PasswordRules.ONE_SYMBOL_CHARACTER);
+        Set<RegistrationConfiguration.PasswordRules> numberMandatory =
+            Set.of(RegistrationConfiguration.PasswordRules.ONE_NUMBER_CHARACTER);
+        Set<RegistrationConfiguration.PasswordRules> upperAndNumberMandatory =
+            Set.of(
+                RegistrationConfiguration.PasswordRules.ONE_UPPER_CASE_CHARACTER,
+                RegistrationConfiguration.PasswordRules.ONE_NUMBER_CHARACTER
+            );
+        Set<RegistrationConfiguration.PasswordRules> lowerAndUpperMandatory =
+            Set.of(
+                RegistrationConfiguration.PasswordRules.ONE_LOWER_CASE_CHARACTER,
```

Comment:
Possibly just me but I tend to use static import for this kind of things, to make the method body shorter and easier to read (i.e., getting rid of the long `RegistrationConfiguration.PasswordRules.` everywhere).

Reply:
It's a matter of taste, usually I prefer to see the where the constant is coming from on my side, even if it's longer. A bit less of a problem in a test though...